velox/functions/prestosql/tests/ArrayShuffleTest.cpp

```diff
@@ -0,0 +1,316 @@
+        evaluate<ArrayVector>("shuffle(C0)", makeRowVector({input}));
+
+    DecodedVector decodedActual;
+    decodedActual.decode(*actualVector.get());
+    exec::VectorReader<Array<int32_t>> readerActual(&decodedActual);
+
+    for (auto i = 0; i < actualVector->size(); i++) {
+      auto actualArray = read<int32_t, false>(readerActual, i).materialize();
+      auto expectedArray =
+          read<int32_t, false>(readerExpected, i).materialize();
+
+      ASSERT_TRUE(std::is_permutation(
+          actualArray.begin(), actualArray.end(), expectedArray.begin()))
+          << "Actual array " << printArray(actualArray) << " at " << i
+          << " must produce a permutation of expected array "
+          << printArray(expectedArray);
+
+      distinctValueSet.insert(actualArray);
+    }
+
+    // Shuffled arrays should be X% different in total.
+    const int32_t kThreshold =
+        (int32_t)(actualVector->size() * kUniquenessRate);
+    auto numDistinctValues = distinctValueSet.size();
+    ASSERT_TRUE(numDistinctValues >= kThreshold)
+        << "Shuffle " << input->encoding()
+        << " array must yield >= " << kThreshold
+        << " distinct values, but only got " << numDistinctValues;
+  }
+};
+} // namespace
+
+TEST_F(ArrayShuffleTest, normalArrays) {
+  auto input = makeNullableArrayVector<int64_t>(
+      {{},
+       {std::nullopt},
+       {std::nullopt, std::nullopt},
+       {-1, 0, 1},
+       {std::nullopt, 0, 0},
+       {std::numeric_limits<int64_t>::max(),
+        std::numeric_limits<int64_t>::min(),
+        0,
+        1,
+        2,
+        3}});
+  testShuffle<int64_t, true>(input);
+}
+
+TEST_F(ArrayShuffleTest, nestedArrays) {
+  using innerArrayType = std::vector<std::optional<int64_t>>;
+  using outerArrayType =
+      std::vector<std::optional<std::vector<std::optional<int64_t>>>>;
+  innerArrayType a{1, 2, 3, 4};
+  innerArrayType b{5, 6};
+  innerArrayType c{6, 7, 8};
+  outerArrayType row1{{a}, {b}};
+  outerArrayType row2{std::nullopt, std::nullopt, {a}, {b}, {c}};
+  outerArrayType row3{{{}}};
+  outerArrayType row4{{{std::nullopt}}};
+  auto input =
+      makeNullableNestedArrayVector<int64_t>({{row1}, {row2}, {row3}, {row4}});
+
+  testShuffle<Array<int64_t>, true>(input);
+}
+
+TEST_F(ArrayShuffleTest, sortAndShuffle) {
+  auto input = makeNullableArrayVector<int64_t>(
+      {{-1, 0, std::nullopt, 1, std::nullopt},
+       {4, 1, 5, 3, 2},
+       {std::numeric_limits<int64_t>::max(),
+        std::numeric_limits<int64_t>::min(),
+        4,
+        1,
+        3,
+        2,
+        std::nullopt}});
+  auto inputVector = makeRowVector({input});
+  auto result1 = evaluate<ArrayVector>("array_sort(C0)", inputVector);
+  auto result2 = evaluate<ArrayVector>("array_sort(shuffle(C0))", inputVector);
+
+  assertEqualVectors(result1, result2);
+}
+
+TEST_F(ArrayShuffleTest, constantEncoding) {
+  vector_size_t size = 100;
+  // Considering the cases of empty array, nullable array, duplicate-element
```

Comment:
typo: "Considering the cases" -> "Test empty array, array with null element and array with duplicate elements."